bolides/bdf.py

```diff
@@ -0,0 +1,1032 @@
+        n : int
+            Number of closest detections to return.
+
+        Returns
+        -------
+        bdf : `~BolideDataFrame`
+            The filtered data.
+
+        """
+
+        dt = datetime.fromisoformat(datestr)
+        return self.iloc[(self['datetime'] - dt).abs().argsort()].head(n)
+
+    def get_closest_by_loc(self, lon, lat, n=1):
+        """Get the n bolides closest to a given longitude and latitude
+
+        Parameters
+        ----------
+        lon, lat : int
+            Longitude and latitude to search for bolides near.
+        n : int
+            Number of closest detections to return.
+
+        Returns
+        -------
+        bdf : `~BolideDataFrame`
+            The filtered data.
+
+        """
+        lon_diff = (self['longitude'] - lon).abs()
+        lat_diff = (self['latitude'] - lat).abs()
+        tot_diff = lon_diff + lat_diff
+        return self.iloc[tot_diff.argsort()].head(n)
+
+    def filter_boundary(self, boundary, intersection=False, interior=True):
+        """Filter data to only points within specified boundaries.
+
+        Parameters
+        ----------
+        boundary: str or list of str
+            The boundary (boundaries) to filter the `~BolideDataFrame` by.
+            Refer to `~bolides.fov_utils.get_boundary`.
+        intersection: bool
+            - True: keep detections within the union of all boundaries given.
+            - False: keep detections within the intersection of all boundaries given.
+        interior: bool
+            - True: default behavior.
+            - False: keep detections outside the union or intersection of the boundaries, instead of inside.
+
+        Returns
+        -------
+        `~BolideDataFrame`
+            The filtered `~BolideDataFrame`
+        """
+
+        import pyproj
+        # define Azimuthal Equidistant projection
+        aeqd = pyproj.Proj(proj='aeqd', ellps='WGS84', datum='WGS84', lat_0=90, lon_0=0).srs
+
+        # put bdf into the correct CRS
+        crs = self.geometry.crs
+        bdf = self.to_crs(aeqd)
+
+        from .fov_utils import get_boundary
+        final_polygon = get_boundary(boundary, intersection=intersection, collection=False)
+
+        # clip with the polygon, which is in Azimuthal Equidistant
+        points_in_poly = [pt.within(final_polygon) for pt in bdf['geometry']]
+        if interior:
+            bdf = bdf[points_in_poly]
+        else:
+            bdf = bdf[~points_in_poly]
+        # project bdf back to original CRS
+        bdf = bdf.to_crs(crs)
+
+        force_bdf_class(bdf)
+
+        return bdf
+
+    def filter_observation(self, sensors, intersection=False):
+        """Filter data to only points observable (in time and space) by a given sensor.
+
+        Parameters
+        ----------
+        sensors: str or list of str
+            The sensor(s) to filter the `~BolideDataFrame` by:
+
+            - ``'glm16'``: The Geostationary Lightning Mapper aboard GOES-16.
```

Comment:
The detectedBy strings are 'GLM-16' and 'GLM-17' but here the filter strings are 'glm16' and 'glm17'. We should standardize the nomenclature, so should use the same strings as in the website data. We could also have short-hand options. I like 'G16' and 'G17' as short-hand.

Reply:
Implemented this in 7f49f433a12f5e91f535ef1069465bc24c20f665. The input is case-insensitive so something like 'g16' (three key-presses instead of four!) works too.

Reply:
Tested, and works.